xapian-core/tests/api_queryparser.cc

```diff
@@ -1725,6 +1725,84 @@ DEFINE_TESTCASE(qp_range4, !backend) {
     return true;
 }
 
+static const test test_value_filesizerange1_queries[] = {
+    { "10K..20K", "0 * VALUE_RANGE 1 10240 20480" },
+    { "10..50M", "0 * VALUE_RANGE 1 10485760 52428800" },
+    { "10K..20", "Unknown range operation" },
+    { "10A..20A", "Unknown range operation" },
+    { "10..20H", "Unknown range operation" },
+    { NULL, NULL }
+};
+
+// Test FileSizeRangeProcessor
+DEFINE_TESTCASE(qp_filesizerange1, !backend) {
+    Xapian::QueryParser qp;
+    Xapian::FileSizeRangeProcessor rp_filesize(1);
+    qp.add_rangeprocessor(&rp_date);
+    for (const test *p = test_value_filesizerange1_queries; p->query; ++p) {
+	string expect, parsed;
+	if (p->expect)
+	    expect = p->expect;
+	else
+	    expect = "parse error";
+	try {
+	    Xapian::Query qobj = qp.parse_query(p->query);
+	    parsed = qobj.get_description();
+	    expect = string("Query(") + expect + ')';
+	} catch (const Xapian::QueryParserError &e) {
+	    parsed = e.get_msg();
+	} catch (const Xapian::Error &e) {
+	    parsed = e.get_description();
+	} catch (...) {
+	    parsed = "Unknown exception!";
+	}
+	tout << "Query: " << p->query << '\n';
+	TEST_STRINGS_EQUAL(parsed, expect);
+    }
+    return true;
+}
+
+static const test test_value_filesizerange2_queries[] = {
+    { "size:10..20K", "0 * VALUE_RANGE 1 10240 20480" },
+    { "size:10..50M", "0 * VALUE_RANGE 1 10485760 52428800" },
+    { "filesize:10B..1M", "0 * VALUE_RANGE 2 10 1048576" },
+    { "file-size:10A..20M", "Unknown range operation" },
+    { "size:10K..100", "Unknown range operation" },
+    { NULL, NULL }
+};
+
+// Feature test FileSizeRangeProcessor with prefixes 
+DEFINE_TESTCASE(qp_filesizerange2, !backend) {
+    Xapian::QueryParser qp;
+    Xapian::FileSizeRangeProcessor rp_filesize_1(1, "size:");
+    Xapian::FileSizeRangeProcessor rp_filesize_2(2, "filesize:");
+    Xapian::FileSizeRangeProcessor rp_filesize_3(3, "file-size:");
+    qp.add_rangeprocessor(&rp_filesize_1);
+    qp.add_rangeprocessor(&rp_filesize_2);
+    qp.add_rangeprocessor(&rp_filesize_3);
```

Comment:
I think it makes sense to just merge the testcases from `qp_filesizerange1` into those tests here.  We'll get better coverage for the time spent running tests that way, as the difference between the single RP and multiple RP case isn't interesting.